clients/venice-client/src/main/java/com/linkedin/venice/fastclient/MultiKeyRequestContext.java

```diff
@@ -167,6 +169,14 @@ public void setRoutesForPartitionMapping(Map<Integer, Set<String>> routesForPart
     this.routesForPartition = routesForPartition;
   }
 
+  public void setFanoutSize(int fanoutSize) {
+    this.fanoutSize = fanoutSize;
+  }
+
+  public int getFanoutSize() {
+    return fanoutSize;
+  }
```

Comment:
Nit,
do we need to maintain `fanoutSize` field here?
It seems we can directly use `routeRequests.size()` as the fanout size, right?

Reply:
That was my initial implementation but that didn't work because in the `RetriableAvroGenericStoreClient` we decided to pass in a new request context instead of the original which is what the metric emission will use. This is why I added the field and setter. Alternatively we need to expose setting the `routeRequests` which is a more intrusive change IMO.
```
 requestContext.setFanoutSize(originalRequestContext.getFanoutSize());
```